internal/chooserbenchmark/visualization.go

```diff
@@ -0,0 +1,330 @@
+		}
+		if server.counter < minRequestCount {
+			minRequestCount = server.counter
+		}
+		groupName := server.groupName
+		if _, ok := serversByGroup[groupName]; !ok {
+			serverGroupNames = append(serverGroupNames, groupName)
+			serversByGroup[groupName] = make([]*Server, 0)
+		}
+		serversByGroup[groupName] = append(serversByGroup[groupName], server)
+	}
+	sort.Strings(serverGroupNames)
+
+	buckets := make([]int, serverHistogramBuckets)
+	diff := maxRequestCount - minRequestCount
+	width := diff / (serverHistogramBuckets - 1)
+	if width == 0 {
+		width = 1
+	}
+	cur := minRequestCount
+	for i := 0; i < serverHistogramBuckets; i++ {
+		cur += width
+		buckets[i] = cur
+	}
+
+	serverData := make(map[string]*ServerGroupMeta)
+	for _, groupName := range serverGroupNames {
+		meta := NewServerGroupMeta(groupName, serversByGroup[groupName], buckets)
+		count := len(meta.servers)
+		if count > maxServerCount {
+			maxServerCount = count
+		}
+		for _, val := range meta.histogram {
+			if val > maxHistogramVal {
+				maxHistogramVal = val
+			}
+		}
+		serverData[groupName] = meta
+	}
+
+	idLen := NormalizeLength(GetNumLength(maxServerCount))
+	counterLen := NormalizeLength(GetNumLength(maxRequestCount))
+	freqLen := NormalizeLength(GetNumLength(maxHistogramVal))
+
+	tableStarLen := displayWidth - idLen - counterLen - 2*tabWidth
+	tableStarUnit := float64(tableStarLen) / float64(maxRequestCount)
+
+	histogramStarLen := displayWidth - counterLen - freqLen - 2*tabWidth
+	histogramStarUnit := float64(histogramStarLen) / float64(maxHistogramVal)
+
+	clients := ctx.Clients
+	clientGroupNames := make([]string, 0)
+	clientsByGroup := make(map[string][]*Client)
+	for _, client := range clients {
+		groupName := client.groupName
+		if _, ok := clientsByGroup[groupName]; !ok {
+			clientGroupNames = append(clientGroupNames, groupName)
+			clientsByGroup[groupName] = make([]*Client, 0)
+		}
+		clientsByGroup[groupName] = append(clientsByGroup[groupName], client)
+	}
+	sort.Strings(clientGroupNames)
+
+	clientData := make(map[string]*ClientGroupMeta)
+	maxLatencyFrequency := 0
+	for _, groupName := range clientGroupNames {
+		meta := NewClientGroupMeta(groupName, clientsByGroup[groupName])
+		for _, freq := range meta.histogram {
+			if freq > maxLatencyFrequency {
+				maxLatencyFrequency = freq
+			}
+		}
+		clientData[groupName] = meta
+	}
+
+	latencyStarUnit := float64(latencyHeight) / float64(maxLatencyFrequency)
+
+	return &Visualizer{
+		clientGroupNames: clientGroupNames,
+		clientData:       clientData,
+		serverGroupNames: serverGroupNames,
+		serverData:       serverData,
+		buckets:          buckets,
+		config: &DisplayConfig{
+			idLen:             idLen,
+			freqLen:           freqLen,
+			counterLen:        counterLen,
+			tableStarUnit:     tableStarUnit,
+			histogramStarUnit: histogramStarUnit,
+			latencyStarUnit:   latencyStarUnit,
+		},
+	}
+}
+
+func (sgm *ServerGroupMeta) ServerGroupVisualization(vis *Visualizer) {
```

Comment:
Let's adda verb to the method since it's doing something: `VisualizeServerGroup`